### PR TITLE
Feature/graveler revert implementation

### LINKED ISSUE #1184: Graveler Revert implementation
The graveler implementation lake revert implementation. The current MVCC based implementation discards the changes until the commit, as long as there is no branching out before the commit.

### PR DESCRIPTION
**Pull request by johnnyaug**
Closes #1184

### CODE REVIEW

File: catalog/rocks/entry_catalog.go
Comment (nopcoder) [inline catalog/rocks/entry_catalog.go:346]: - Use ValidateRef for `ref`
- Use ValidateRequiredString for `committer` and `message`

File: graveler/graveler.go
Comment (itaiad200) [inline graveler/graveler.go:1060]: Time is missing.

File: docs/reference/commands.md
Comment (itaiad200) [inline docs/reference/commands.md:103]: I would add an explicit example.

File: cmd/lakectl/cmd/branch.go
Comment (itaiad200) [inline cmd/lakectl/cmd/branch.go:127]:
```suggestion
	Short: "given a commit, record a new commit to reverse the effect of this commit",
```

File: catalog/rocks/fake_graveler_test.go
Comment (itaiad200) [inline catalog/rocks/fake_graveler_test.go:158]: Anonymise all params.

```suggestion
func (g *FakeGraveler) Revert(_ context.Context, _ graveler.RepositoryID, _ graveler.BranchID, _ graveler.Ref, _, _ string, _ graveler.Metadata) (graveler.CommitID, error) {
```

File: catalog/rocks/cataloger.go
Comment (itaiad200) [inline catalog/rocks/cataloger.go:487]: Why use default committer and not pass the actual executor of the action?

File: graveler/graveler.go
Comment (itaiad200) [inline graveler/graveler.go:1014]: I don't see where you handle a request to revert `C0` which isn't in the branch.

File: api/api_controller.go
Comment (itaiad200) [inline api/api_controller.go:1332]: You're using the Revert permissions here on purpose?

File: api/api_controller.go
Comment (itaiad200) [inline api/api_controller.go:1318]: `db.ErrNotFound` ?

**Review by itaiad200 — COMMENTED** [review]
Very nice! 
Few minor comments.

File: api/api_controller.go
Comment (nopcoder) [inline api/api_controller.go:1318]: this will work for both implementations?

File: api/client.go
Comment (nopcoder) [inline api/client.go:496]: Use NewResetBranchParams() ...

File: catalog/mvcc/cataloger_rollback_commit.go
Comment (nopcoder) [inline catalog/mvcc/cataloger_rollback_commit.go:12]: add logging in case we will need to know

File: cmd/lakectl/cmd/branch.go
Comment (nopcoder) [inline cmd/lakectl/cmd/branch.go:127]: without the 'with'

File: cmd/lakectl/cmd/branch.go
Comment (nopcoder) [inline cmd/lakectl/cmd/branch.go:139]: optional I guess

File: docs/branching/recommendations.md
Comment (nopcoder) [inline docs/branching/recommendations.md:27]: general note - this will be published as soon as you merge.

File: docs/reference/commands.md
Comment (nopcoder) [inline docs/reference/commands.md:102]: "commit ref" -> align with the usage later "commit ID" pick one

File: graveler/graveler.go
Comment (nopcoder) [inline graveler/graveler.go:1049]: can't log the parents[0] at this point - you wanted the branch id maybe?

File: graveler/graveler.go
Comment (nopcoder) [inline graveler/graveler.go:1041]: Added .Ref() you can use to make the syntax cooler

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: api/api_controller.go
Comment (johnnyaug) [reply api/api_controller.go:1318]: good catch

File: api/api_controller.go
Comment (johnnyaug) [reply api/api_controller.go:1332]: Yes, didn't want to add an extra set of permissions

File: graveler/graveler.go
Comment (johnnyaug) [reply graveler/graveler.go:1032]: synced f2f - the behavior is similar to git, and will apply the reverse change regardless if the commit is an ancestor of the branch head.

File: api/api_controller.go
Comment (nopcoder) [inline api/api_controller.go:1328]: need to check both errors until we remove the mvcc

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: api/api_controller.go
Comment (johnnyaug) [reply api/api_controller.go:1328]: This flow is not supported in MVCC

**Review by nopcoder — APPROVED** [review]

**Review by itaiad200 — APPROVED** [review]

**Pull request merged**
